allensdk/brain_observatory/behavior/data_objects/running_speed/running_speed.py

```diff
@@ -0,0 +1,179 @@
+
+import json
+from typing import Optional
+
+from cachetools import cached, LRUCache
+from cachetools.keys import hashkey
+
+import pandas as pd
+
+from pynwb import NWBFile, ProcessingModule
+from pynwb.base import TimeSeries
+
+from allensdk.core.exceptions import DataFrameIndexError
+from allensdk.internal.api import PostgresQueryMixin
+from allensdk.brain_observatory.behavior.data_objects import (
+    DataObject, StimulusTimestamps
+)
+from allensdk.brain_observatory.behavior.data_files import (
+    StimulusFile
+)
+from allensdk.brain_observatory.behavior.data_objects.running_speed.running_processing import (  # noqa: E501
+    get_running_df
+)
+
+
+def from_json_cache_key(
+    cls, dict_repr: dict, filtered: bool, zscore_threshold: float
+):
+    return hashkey(json.dumps(dict_repr), filtered, zscore_threshold)
+
+
+def from_lims_cache_key(
+    cls, db, behavior_session_id: int, ophys_experiment_id: Optional[int],
+    filtered: bool, zscore_threshold: float
+):
+    return hashkey(
+        behavior_session_id, ophys_experiment_id, filtered, zscore_threshold
+    )
+
+
+class RunningSpeed(DataObject):
```

Comment:
can you add a high level docstring to the data objects? For example, to help a dev understand quickly, "what is the difference between `RunningSpeed` and `RunningAcquisition`?

Reply:
Done